model/transformer.py

```diff
@@ -0,0 +1,156 @@
+"""
+  Model Class for Transformer
+"""
+
+import math
+import torch
+import torch.nn as nn
+
+from torch.nn import TransformerDecoder, TransformerDecoderLayer, TransformerEncoder, TransformerEncoderLayer
+from yacs.config import CfgNode
+
+
+class PositionalEncoding(nn.Module):
+    """
+    Positional encoding for transformer model
+    Code from https://pytorch.org/tutorials/beginner/transformer_tutorial.html
+    """
+
+    def __init__(self,
+                 d_model: int,
+                 dropout=0.1,
+                 max_len=500):
+        super(PositionalEncoding, self).__init__()
+        self.dropout = nn.Dropout(p=dropout)
+
+        positional_encoding = torch.zeros(max_len, d_model)
+        position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
+        div_term = torch.exp(torch.arange(
+            0, d_model, 2).float() * (-math.log(10000.0) / d_model))
+        positional_encoding[:, 0::2] = torch.sin(position * div_term)
+        positional_encoding[:, 1::2] = torch.cos(position * div_term)
+        positional_encoding = positional_encoding.unsqueeze(0).transpose(0, 1)
+        self.register_buffer('pe', positional_encoding)

```

Comment:
if you took this code from a blog post or something it would be best to link it in a comment (maybe here, maybe at the top of the file)